### PR TITLE
Support build and release of diffs of the host-spec on the CI

### PR DESCRIPTION
**Pull request by FlorianFranzen**
This PR adds command line level diff support to the host-spec and builds them for pull requests and releases automatically.

### CODE REVIEW

**Comment by lamafab** [comment]
@lamafab @drskalman review this and approve PR.

**Comment by FlorianFranzen** [comment]
I created a demo of what files will be added to a release with this Code by the CI [here](https://github.com/FlorianFranzen/polkadot-spec/releases).

I also went through the research website git logs and assign incrementing numbers to each version release on the website. I was forced to tag the folder restructuring as a release, because that is the last commit the code can currently compare to.

File: host-spec/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+host-spec.tex
```
Comment (lamafab) [inline host-spec/.gitignore:1]: Might want to remove this, otherwise LGTM.

**Review by lamafab — APPROVED** [review]

File: host-spec/.gitignore (hunk 1 above)
Comment (lamafab) [reply host-spec/.gitignore:1]: Also, maybe add newlines between each CI step (in the yaml file).

Comment (FlorianFranzen) [reply host-spec/.gitignore:1]: I will see if the .tex files are still generated correctly and make a decision based on that. The yaml formatting can be changed, but if we do so, we should change it in all workflow files.

**Comment by FlorianFranzen** [comment]
I noticed that one thing left to do is to install the algorithm package on out CI. I don't think it renders them correctly right now.

**Comment by FlorianFranzen** [comment]
> I noticed that one thing left to do is to install the algorithm package on out CI. I don't think it renders them correctly right now.

Ok, this was fixed with f13d670. I also rebased the branch on the latest master, to avoid any merge conflicts for the other CI work I am currently doing. 

@drskalman Did you find time to review this yet?

**Review by drskalman — APPROVED** [review]
Awesome! Algorithms are also working. I'd personally use it to check Spec updates etc. I assume implementers are going to use it beside checking the CI version to see what has changed when. I added few words to the README for people who don't know what to look for. Thanks a lot.

**Comment by drskalman** [comment]
Just checking if I can comment on a pull request :-P

**Pull request merged**